version_control.go

```diff
@@ -233,50 +235,53 @@ func (c *VersionCache) UpdateBinary(ctx context.Context, binary string) (bool, e
 	// this is a new version
 	c.logger.Infof("new version (%s) found for %s", verData.Version, binary)
 
-	// download and record the sha of the download itself
-	var err error
-	verData.DlPath, err = utils.DownloadFile(ctx, verData.URL, c.logger)
-	if err != nil {
-		if isCustomURL {
-			data.brokenTarget = true
+	if !goodBytes {
```

Comment:
The bulk of this fix is just moving a lot of stuff (that should only happen when a redownload IS needed) in this block, and re-ordering one or two other snippets. Nearly zero "new" code.